lib/schemes/HttpUriPlugin.js

```diff
@@ -612,9 +603,30 @@ class HttpUriPlugin {
 									res.statusCode >= 301 &&
 									res.statusCode <= 308
 								) {
-									return finishWith({
+									const result = {
 										location: new URL(location, url).href
-									});
+									};
+									if (
+										!cachedResult ||
+										!("location" in cachedResult) ||
+										cachedResult.location !== result.location ||
+										cachedResult.validUntil < validUntil ||
+										cachedResult.storeLock !== storeLock ||
+										cachedResult.storeCache !== storeCache ||
+										cachedResult.etag !== etag
+									) {
+										return finishWith(result);
+									} else {
+										logger.debug(`GET ${url} [${res.statusCode}] (unchanged)`);
+										return callback(null, {
```

Comment:
is this for catching infinite redirect?

Reply:
since there is no 304 response for a cached redirect, we compare location to check if the response is actually different and only update the cache when it is

Reply:
hm.. but what a reason to store only partial response? `finishWith` stores in cache not original `url` as key. I could be wrong, but it looks like:
- we store just ({ location }) in store
- finishing with new `url` updating/store for new cache entry

Reply:
there is
```js
if ("location" in result) {
								return resolveContent(
									result.location,
									integrity,
									(err, innerResult) => {
										if (err) return callback(err);
										callback(null, {
											entry: innerResult.entry,
											content: innerResult.content,
											storeLock: innerResult.storeLock && result.storeLock
										});
									}
								);
```
that updates/store new entry for new `url`

Reply:
ok, I got the idea..